xray/src/generation.rs

```diff
@@ -715,3 +674,56 @@ pub fn build_xray_quadtree(
 
     Ok(())
 }
+
+fn build_level(
+    output_directory: &Path,
+    tile: &Tile,
+    current_level: u8,
+    previous_level_nodes: &FnvHashSet<NodeId>,
+    parameters: &XrayParameters,
+) -> FnvHashSet<NodeId> {
+    let current_level_nodes: FnvHashSet<NodeId> = previous_level_nodes
+        .iter()
+        .filter_map(|node| node.parent_id())
+        .collect();
```

Comment:
I wonder whether it would be easier to understand, if we pulled out the computation of `current_level_nodes` into the for loop and passed this instead of `previous_level_nodes`. Then the function wouldn't need to return anything.

Reply:
I pulled it out and I think it looks better?